### PR TITLE
Use Prerelease System.Text.Json

### PR DESCRIPTION
**Pull request by crobibero**
Switch back to stable releases once .net5 is released.

### CODE REVIEW

File: Jellyfin.Data/Jellyfin.Data.csproj [hunk 1]
```diff
@@ -22,6 +22,7 @@
   <ItemGroup>
     <PackageReference Include="Microsoft.EntityFrameworkCore.Relational" Version="3.1.6" />
     <PackageReference Include="Microsoft.EntityFrameworkCore.Sqlite" Version="3.1.6" />
+    <PackageReference Include="System.Text.Json" Version="5.0.0-preview.7.20364.11" />
```
[the remark this answers is not in the document]
Comment (crobibero) [reply Jellyfin.Data/Jellyfin.Data.csproj:25]: If I didn't add it explicitly it was still implicitly using the latest stable

File: Emby.Server.Implementations/HttpServer/WebSocketConnection.cs [hunk 2]
```diff
@@ -209,6 +209,12 @@ private async Task ProcessInternal(PipeReader reader)
                 return;
             }
 
+            if (stub is null)
+            {
+                _logger.LogError("Error processing web socket message");
```
Comment (Bond-009) [inline Emby.Server.Implementations/HttpServer/WebSocketConnection.cs:214]: This should never happen AFAIK?

File: Jellyfin.Api/Helpers/MediaInfoHelper.cs [hunk 3]
```diff
@@ -127,7 +127,11 @@ public class MediaInfoHelper
             {
                 // Since we're going to be setting properties on MediaSourceInfos that come out of _mediaSourceManager, we should clone it
                 // Should we move this directly into MediaSourceManager?
-                result.MediaSources = JsonSerializer.Deserialize<MediaSourceInfo[]>(JsonSerializer.SerializeToUtf8Bytes(mediaSources));
+                var mediaSourcesClone = JsonSerializer.Deserialize<MediaSourceInfo[]>(JsonSerializer.SerializeToUtf8Bytes(mediaSources));
+                if (mediaSourcesClone != null)
```
Comment (Bond-009) [inline Jellyfin.Api/Helpers/MediaInfoHelper.cs:131]: When would this be `null`?

File: Jellyfin.Server/Migrations/Routines/MigrateDisplayPreferencesDb.cs [hunk 4]
```diff
@@ -81,6 +81,11 @@ public void Perform()
                 foreach (var result in results)
                 {
                     var dto = JsonSerializer.Deserialize<DisplayPreferencesDto>(result[3].ToString(), _jsonOptions);
+                    if (dto is null)
```
Comment (Bond-009) [inline Jellyfin.Server/Migrations/Routines/MigrateDisplayPreferencesDb.cs:84]: We should stay consistent, everywhere else uses `==` and `!=`

**Review by Bond-009 — CHANGES_REQUESTED** [review]

File: Emby.Server.Implementations/HttpServer/WebSocketConnection.cs (hunk 2 above)
Comment (crobibero) [reply Emby.Server.Implementations/HttpServer/WebSocketConnection.cs:214]: in .net5 the jsonserializer always returns nullable objects

**Comment by crobibero** [comment]
Many JsonConverters are no longer needed- tested by posting a test class:
```c#
public class TestDto
    {
        public int IntValue { get; set; }

        public int? NullableIntValue { get; set; }

        public double DoubleValue { get; set; }

        public double? NullableDoubleValue { get; set; }

        public long LongValue { get; set; }

        public long? NullableLongValue { get; set; }

        public Guid GuidValue { get; set; }

        public Guid? NullableGuidValue { get; set; }

        public Dictionary<ImageType, string> ImageTypeDictionary { get; set; }
    }
```

Input:
```json
{
  "IntValue": "0",
  "NullableIntValue": 1,
  "DoubleValue": "0",
  "NullableDoubleValue": null,
  "LongValue": "0",
  "NullableLongValue": null,
  "GuidValue": "3fa85f6457174562b3fc2c963f66afa6",
  "NullableGuidValue": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
  "ImageTypeDictionary": {
    "Primary": "string",
    "Art": "string",
    "Backdrop": "string",
    "Banner": "string",
    "Logo": "string",
    "Thumb": "string",
    "Disc": "string",
    "Box": "string",
    "Screenshot": "string",
    "Menu": "string",
    "Chapter": "string",
    "BoxRear": "string",
    "Profile": "string"
  }
}
```

Output:
```json
{
  "IntValue": 0,
  "NullableIntValue": 1,
  "DoubleValue": 0,
  "LongValue": 0,
  "GuidValue": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
  "NullableGuidValue": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
  "ImageTypeDictionary": {
    "Primary": "string",
    "Art": "string",
    "Backdrop": "string",
    "Banner": "string",
    "Logo": "string",
    "Thumb": "string",
    "Disc": "string",
    "Box": "string",
    "Screenshot": "string",
    "Menu": "string",
    "Chapter": "string",
    "BoxRear": "string",
    "Profile": "string"
  }
}
```

File: Emby.Server.Implementations/HttpServer/WebSocketConnection.cs [hunk 5]
```diff
@@ -209,6 +209,12 @@ private async Task ProcessInternal(PipeReader reader)
                 return;
             }
 
+            if (stub is null)
```
Comment (Bond-009) [inline Emby.Server.Implementations/HttpServer/WebSocketConnection.cs:212]:
```suggestion
            if (stub == null)
```

File: Jellyfin.Server/Migrations/Routines/MigrateUserDb.cs [hunk 6]
```diff
@@ -74,7 +74,12 @@ public void Perform()
 
                 foreach (var entry in queryResult)
                 {
-                    UserMockup mockup = JsonSerializer.Deserialize<UserMockup>(entry[2].ToBlob(), JsonDefaults.GetOptions());
+                    UserMockup? mockup = JsonSerializer.Deserialize<UserMockup>(entry[2].ToBlob(), JsonDefaults.GetOptions());
+                    if (mockup is null)
```
Comment (Bond-009) [inline Jellyfin.Server/Migrations/Routines/MigrateUserDb.cs:78]:
```suggestion
                    if (mockup == null)
```

**Review by Bond-009 — CHANGES_REQUESTED** [review]

**Review by Bond-009 — APPROVED** [review]

**Review by barronpm — APPROVED** [review]

**Pull request merged**